tools/PI/DevHome.PI/ViewModels/InsightsPageViewModel.cs

```diff
@@ -61,17 +53,19 @@ private void TargetApp_PropertyChanged(object? sender, PropertyChangedEventArgs
 
     internal void AddInsight(Insight insight)
     {
+        var barWindow = Application.Current.GetService<PrimaryWindow>().DBarWindow;
```

Comment:
>; [](http://example.com/codeflow?start=82&length=1)

Ugh... all this feels wrong. One page's view model feeding data to another page's view model?

Feels like there should be an Insight Service, that exposes an Observable collection that both the Insights and Bar view models would subscribe to.

I know that you didn't create this originally... mind fixing this up?